### PR TITLE
Adam's species rework

### LINKED ISSUE #850: repost of my old species rework thing X-The-Dark please add this to projects
null

### PR DESCRIPTION
**Pull request by Cruix**
Closes #850

Phytosians
- [x] Fire multiplier to x2
- [x] Chemical effects
- [x] Extra acid weakness
- [x] Heal on being flashed
- [x] No healing below 150K / above 400K
- [x] Slight speed nerf / ban from head roles
- [x] Thrall effects
- [x] Pheromone speak

Preternis
- [x] Feed off any electricity
- [x] Vulnerable to electric shock
- [x] Immune to radiation / clone / toxin
- [x] Chemical metabolism
- [x] Chemical effects
- [x] Damage modifiers
- [x] Heal with welders / cable coils / screwdrivers
- [x] Speed buff
- [x] Extra environmental temperature damage
- [x] Easy insert implants
- [x] Reject unique organs
- [x] Metal gibs
- [x] Virus effects
- [x] Trinary speak
- [x] Emag effects
- [x] Weak to flashes

Unathi
- [x] Cold effects
- [x] Resistance to heat
- [x] Reduced electric shock damage
- [x] Pressure resistance
- [x] Speed nerf
- [x] Extra stamina damage / slower recovery
- [x] Special eyes
- [x] Vulnerable to radiation

### CODE REVIEW

**Comment by Cruix** [comment]
This is the third time I have coded this. And I have become exceedingly efficient at it.

**Comment by Super3222** [comment]
:+1: x :100:
This PR is golden.

**Comment by Time-Green** [comment]
Isn't doubla damage a bit high on phytosians now? You could already completely annihalate them with it, and they got nerfed real hard here.
Can we keep it on 50%?

**Comment by ghost** [comment]
fuck off time green

**Comment by ShadowDeath6** [comment]
> preternis completely immune to radbow

Fuck off. Immunity to any damage is OP, make it .7 or something.

**Comment by Cruix** [comment]
Bear in mind that they cannot heal from toxin damage, at all.

**Comment by ShadowDeath6** [comment]
they also can't TAKE toxin damage

**Comment by Super3222** [comment]
@ShadowDeath6  Being immune to some damage isn't too bad and, some races are like that. I do see where you're coming from, but I still suggest making it 0.2-0.4 instead of 0.7. With that idea, they're more like androids and lean over to the idea of being half borg. In fact, I'm pretty sure borgs are completely immune to toxin damage.

Just take a look at the other weaknesses they have. If a traitor only relies on "shoot, shoot, kill, kill" with a radbow they deserve to get taken down by the preternis HoS. People will have to come up with strategies and, plan shit out. For instance, people already strategize against phytosian opponents with fire-based weaponry.

**Comment by ShadowDeath6** [comment]
notice how no race is immune to revolver bullets

**Comment by Cruix** [comment]
I'll probably add the target's race to the info a traitor gets on their target, in addition to name and job.

**Comment by ghost** [comment]
boozepower isn't 1-5 anymore

it uses a different method of calc, 0-100

**Comment by Super3222** [comment]
Please, don't forget a changelog. This is gonna be pretty good.

**Comment by Super3222** [comment]
After this gets merged and tested, I do want to look back on the fireloss Phytosians are recieving. As much as I like it, there's a possibility it may weigh too much.

**Comment by Time-Green** [comment]
Like this, a welder does more damage then an esword

**Comment by Time-Green** [comment]
M E R G E

**Comment by Super3222** [comment]
Changelog

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 1]
```diff
@@ -52,21 +52,96 @@
...
+		heal_overall_damage(0, -amount, application)
+
+/mob/living/carbon/human/adjustOxyLoss(amount, updating_health=1, application=DAMAGE_PHYSICAL)
+	if(GODMODE in status_flags)
+		return 0
+	if(dna && dna.species && application != DAMAGE_NO_MULTIPLIER)
+		if(amount > 0)
+			if(application in dna.species.damage_immunities)
+				return
+		else
```
Comment (X-TheDark) [inline code/modules/mob/living/carbon/human/human_damage.dm:78]: This is extremely minor, but do you think adding species level `oxymod` or somesuch variable (for modifying how much oxygen damage the species take (so it's consistent with other `adjustXLoss` procs) would be a lot of work?

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -238,12 +242,13 @@
...
 	//DAMAGE//
+	var/damagemod = (dna && dna.species) ? dna.species.acidmod : 1
 	for(var/obj/item/bodypart/affecting in damaged)
 		affecting.take_damage(acidity, 2*acidity)
 
 		if(affecting.name == "head")
-			if(prob(min(acidpwr*acid_volume/10, 90))) //Applies disfigurement
-				affecting.take_damage(acidity, 2*acidity)
+			if(prob(min(acidpwr*damagemod*acid_volume/10, 90))) //Applies disfigurement
+				affecting.take_damage(acidity*damagemod*damagemod, 2*acidity*damagemod*damagemod)
```
Comment (X-TheDark) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Why is the `damagemod` squared here?

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 3]
```diff
@@ -288,6 +371,64 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
+/datum/species/android/emag_act(mob/living/carbon/human/H, mob/user)
+	if(emagged == 0)
+		emagged = 1
+		if(user)
+			user << "<span class='notice'>You stealthily swipe the cryptographic sequencer along the implants on the back of [H]'s head.</span>"
+		H << "<span class='danger'>You suddenly feel stupid.</span>"
+		H.adjustBrainLoss(30)
+		return 1
+	else if(emagged == 1)
+		emagged = 2
```
Comment (X-TheDark) [inline code/modules/mob/living/carbon/human/species_types.dm:408]: Does `emag_act` have a cooldown? Don't want these things to be insta-magged twice.

File: code/modules/mob/living/carbon/human/human_damage.dm (hunk 1 above)
Comment (Cruix) [reply code/modules/mob/living/carbon/human/human_damage.dm:78]: Nope, should be easy

File: code/modules/mob/living/carbon/human/human_defense.dm (hunk 2 above)
Comment (Cruix) [reply code/modules/mob/living/carbon/human/human_defense.dm]: Probably a mistake with ctrl+H

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 3 above)
Comment (Cruix) [reply code/modules/mob/living/carbon/human/species_types.dm:408]: Same as the normal attack cooldown

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 4]
```diff
@@ -288,6 +371,64 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
 	return SPAN_ROBOT
 
+/datum/species/android/can_accept_organ(mob/living/carbon/human/H, obj/item/organ/O)
+	var/static/list/acceptable = list(/obj/item/organ/heart, /obj/item/organ/lungs, /obj/item/organ/tongue, /obj/item/organ/appendix, /obj/item/organ/cyberimp)
+	return (O.status != ORGAN_ORGANIC || is_type_in_list(O, acceptable))
+
+/datum/species/android/on_gain_disease(mob/living/carbon/human/H, datum/disease/D)
+	D.spread_flags |= AIRBORNE
+
+/datum/species/android/emag_act(mob/living/carbon/human/H, mob/user)
```
Comment (Super3222) [inline code/modules/mob/living/carbon/human/species_types.dm:399]: I thought the original idea was that they had to be lying down to be emagged? Are you sure about giving them straight away brain damage?

**Comment by Ephemeralis** [comment]
You're gonna hate me for this, but maybe tie Preternis bonuses into a set of "organ" implants they start with at roundstart.

A central "controller" unit and then a set of ocular implants, etc, etc. Having the controller unit removes makes all preternis implants stop functioning and periodically deal toxin damage to the person instead.

**Comment by Cruix** [comment]
I'm not averse to that.

**Comment by Time-Green** [comment]
They should violently puke robo gibs for a while

**Comment by ShadowDeath6** [comment]
nobody is going to remove the organs over killing them because it's time consuming, same with re-implanting them because you can already get better stuff from R&D

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 4 above)
Comment (Cruix) [reply code/modules/mob/living/carbon/human/species_types.dm:399]: They can easily fix it with just a screwdriver, so it's not much of a danger unless you want to kill someone and try and mute them with an emag, which is a strategy I would like to see some day.

**Comment by Cruix** [comment]
Updated the wiki pages:
https://wiki.yogstation.net/index.php?title=Phytosian
https://wiki.yogstation.net/index.php?title=Preternis
https://wiki.yogstation.net/index.php?title=Unathi

File: code/modules/mob/living/carbon/human/species_types.dm (hunk 4 above)
Comment (Super3222) [reply code/modules/mob/living/carbon/human/species_types.dm:399]: Hm, I see where you're going with it.

**Comment by Karinrine** [comment]
Is this finished yet?

**Comment by Time-Green** [comment]
I'm going to say that the 2x damage multiplier is extreme.
Every person that knows it exists can kill you in 4 welder hits. Or 3 laser shots.
A welder would do 30 burn damage, that's more then an esword.
A laser does 40. That's the same as the heavy laser gun. Hell, that thing would do 80 burn damage.

If there needs to be an exchange, make phytosians require food, since plants also need more then just lights, and honestly it's sad I can't eat a little without going obese.

Please reconsider the change.

**Comment by ShadowDeath6** [comment]
free infinite salglu is balance enough

**Comment by Time-Green** [comment]
2 hits, you're slowed down and then you can be easily finished, if a single 
punch would kill you, would regen even matter?

**Comment by ghost** [comment]
# FUCK OFF TIME GREEN

apparently me saying it isn't obvious enough

species are intended to be _different_ from humans

Just like plasmamen have **massive, significant downsides** and are thus a hardmode race, phytosians are also a hardmode race in that manner.

They have **significant upsides** where the downside is **being vulnerable as fuck to burn damage**

Yes, you **can** be fourhit by lasers or twohit by heavy lasers or whatever but you also have a lot of upsides to compensate for this one, massive downside

congrats, it's now a hardmode race that is not a straight upgrade to humans like it is currently

d e a l

**Comment by ShadowDeath6** [comment]
I do agree with adam but it's come to my attention that plants are as weak to light beams as shadowlings are

**Comment by ghost** [comment]
good

**Comment by Stealthkibbler** [comment]
So are preternis, you get hit by one ion pulse as preternis and you're basically dicked, besides, HYPE FOR PLANT PEOPLE SPESHUL SNOWFLEK SPEAK. Now we can shit talk all the non cabbages and they wont even know.

**Comment by Dragonforcefive** [comment]
Lol... no reference to the disliked plasma man

**Comment by ShadowDeath6** [comment]
plasmaman wasn't made by us though

**Comment by Dragonforcefive** [comment]
Lol, I meant no one wants to deal with them because no one likes them...

**Comment by Time-Green** [comment]
change burn multiplier back to 1.5x or 1x and this is good to go 👍

**Comment by Super3222** [comment]
> change burn multiplier back to 1.5x or 1x and this is good to go 👍

that's not going to happen unless we get some serious results.

but I'm not going to lie. getting 30 damage from a welder just because you picked a playable race is far from user friendly. It will **most likely** be turned down to 1.5x in the future which might sound like a letdown at first to the phytosian haters until you do the math.

**Comment by Time-Green** [comment]
It currently does 22.5 burn damage, so 2 hits is enough to slow you down and finish you off.
Could we compensate this? Like as long as plants got water in their system the burn damage negates?

**Comment by Dragonforcefive** [comment]
So, what does emagging a Preternis do? And what does the Virus stuff mean?

**Comment by Cruix** [comment]
https://wiki.yogstation.net/index.php?title=Preternis

**Comment by ghost** [comment]
Meanwhile on time-green can't get it through his autistic manchild skull that races have downsides and nobody's gonna cater to his stupid fucking plant name lmao

Deal with it you piece of shit phytos are staying at 2x and if you don't like it go play human or kill yourself

**Comment by Time-Green** [comment]
Can you give a normal reason? You're not open to others suggestions, and you don't know enough to even realize

**Comment by ghost** [comment]
I know the reason.

It's because races have downsides.

that's your reason.

Phytosians will take 2x damage from burn and it will stay that way. You already get your snowflake speak and a utomatic healing, and a bunch of chemical benefits.

You can easily stack a lot of chems to heal out of most things, heal over time, and have snowflake speak. You have a lot of different bonuses to being a phytosian.

The downside is that you are fragile.

Similar to a plasmaman (which is extremely fragile and can't be cloned, which is a hardmode race) phytosians will ALSO be a hardmode race.

Don't like it? Tough.

**Comment by Time-Green** [comment]
Look, you actually backed it up. I respect that.
We'll see the general feedback.

**Comment by drderp3635** [comment]
It would be cool if slimemen had the inverse to preternis where they could self implant biological organs but not mechanical

**Comment by ShadowDeath6** [comment]
Slime girls roundstart when?

**Comment by ghost** [comment]
@Cruix 
please

**Comment by Super3222** [comment]
So I tested the code, and here's my review for Preternis;

Preternis
- Electricity: Yeah... while it's a fun feature to be able to drain electricity from machines - it's rather overpowered. Seriously, at the beginning of the round I walked around, found an APC and began draining it's power. I dropped it's charge to 54%. 

Similarly, it seemed like an alert was raised when I first began to drain an APC's energy.

"NOTICE: the black gloves are interfering with electrical contact - advise removal before activating CONSUME protocol."

Yet, it wasn't interfering with me draining power from the APC.

Eating food brings up something like "NOTICE: Digestive subroutines are inefficient. Seek sustenance via power-cell CONSUME induction." I love it.

- Result of Tesla Revolver:

One bolt. Health bar is red.
I've suffered 63 burn damage.

The thing about this vulnerability is that electricity / shock damage isn't that common. It's a powerful weakness, yes. But even electricity damage isn't as common as some of the stats that Preternis are becoming invulnerable too.

- Speed: Preternis are EXTREMELY fast. If I want to make it to the shuttle in time, than I'll be able to do it from the bar in under 10 seconds compared to a human (and I've been in this situation before) which'd make it 2 seconds late.

It's a pretty dangerous game to play with a pro like this considering speed is pretty important - via running away from security or antags etc etc.

- Organs: It fits really well. The way it works is like this - Preternis hold a cyber implant in their hand, they activate it, and than there's this delay. After the delay, it obviously gets auto injected into their body. You can't "build up" a bunch of implants in the same spot, otherwise the one in that spot will get pushed out. As for the time being I don't see any problems with it.
- Gibbing: I like it. It's basically the same as borgs.
- Flashing: It's an impressive drawback. Flashing will not only blind a preternis, but stun them for around 3-4 ticks.
- Emagging: Perfect if you want to stall a Preternis, but at the same time they can run away just as fast with their speed. Also, when I was reading the code I thought their screen was suppose to get a bunch of LSD like effects - everything looking trippy. This was not the case. Their eyes were only flashing random colors.
- Chemicals: Er... I wasn't getting any damage from drinking an entire flask of holy water. I checked the code, and it seems that Preternis are invulnerable to chemical damage... What

**Comment by ShadowDeath6** [comment]
chems don't metabolize in prets

also why are they weak to shocks if they eat shocks???

also I still don't like that they're entirely immune to radbow

**Comment by Super3222** [comment]
> I wasn't getting any damage from drinking an entire flask of holy water.

I meant unholy water.

> also I still don't like that they're entirely immune to radbow

I somewhat think that radiation damage should be redirected to causing a shortened blind effect, or not.

Preternis are OP right now. We should draw up solutions on how to balance them. My suggestion is make them more vulnerable to brute damage, given that they have amazing agility. Considering that they can heal themselves with welders I don't think it's that much of a problem.

**Comment by Cruix** [comment]
>Electricity: Yeah... while it's a fun feature to be able to drain electricity from machines - it's rather overpowered. Seriously, at the beginning of the round I walked around, found an APC and began draining it's power. I dropped it's charge to 54%.

It's using the same conversion factor as it does right now, APCs just have really small batteries. If you want nutrition fast you can also shock yourself on a door and heal with cable coils, that gives you a pretty generous amount of nutrition.

>Similarly, it seemed like an alert was raised when I first began to drain an APC's energy.
"NOTICE: the black gloves are interfering with electrical contact - advise removal before activating CONSUME protocol."
Yet, it wasn't interfering with me draining power from the APC.

Black gloves have a Siemen's coefficient of less than one, so they make draining power less efficient. You can still do it, it just takes longer. Insulated gloves prevent draining completely. This is mostly flavor, since there is basically no situation where you can't take off your gloves for a second. <sub><sub><sub>(also super secret budget insulated gloves powergaming but shhh you did not hear it from me)</sub></sub></sub>

>Eating food brings up something like "NOTICE: Digestive subroutines are inefficient. Seek sustenance via power-cell CONSUME induction." I love it.

It says that right now

>Emagging: Perfect if you want to stall a Preternis, but at the same time they can run away just as fast with their speed. Also, when I was reading the code I thought their screen was suppose to get a bunch of LSD like effects - everything looking trippy. This was not the case. Their eyes were only flashing random colors.

I'll get to that at some point. The brain damage is pretty crazy, but they can fix it with a screwdriver, so not a big deal.

>Chemicals: Er... I wasn't getting any damage from drinking an entire flask of holy water. I checked the code, and it seems that Preternis are invulnerable to chemical damage... What

That's intended. They don't heal from chemicals either. That's their biggest balance issue.

**Comment by ghost** [comment]
@Cruix 
Merge it  F   UC K KIGNM when

also preternis are supposed to have the chem's effects apply after 40 ticks

like, you know how amantin waits until it all leaves the system then hits you with a fuckton of PAIN

similar to that, in that all reagents in the preternis body are flushed after 40 ticks, regardless of if 10u bicard was added 39 ticks ago and then the pret got 10u kelo.

but they still do the same damage/healing as before

**Comment by Cruix** [comment]
WHAT

**Comment by optimumtact** [comment]
I will ask goof to finish this for you, he loves playing with alien species pr's

**Comment by Cruix** [comment]
Adam figured out how to make labels we are all doomed

**Comment by ShadowDeath6** [comment]
@AdamElTablawy don't make new labels for labels we already have you keem

**Comment by Super3222** [comment]
The virology rework you ported over is conflicting with this PR now.
And lots of other conflicts.

**Comment by drderp3635** [comment]
I feel like this pr is like the block of c4 you set the timer to 9999999999 then plant on the wall as decoration

**Comment by ShadowDeath6** [comment]
zzz...

**Comment by Time-Green** [comment]
aaahhhh wheeen issss it gettting meeeerrggeeeed
_Also change damage multipler back to 1.5_

**Comment by Super3222** [comment]
> Cruix commented on Sep 22, 2016 • edited

**Comment by Time-Green** [comment]
>Cruix commented on 23 Sep 2016 • edited

**Comment by Time-Green** [comment]
DO ITTTTTTTTTTTTTTTTTTT

**Comment by Super3222** [comment]
# CARRY ON MY WAYWARD SON
# FOR THERE'LL BE PEACE WHEN YOU ARE DONE
# LAY YOUR WEARY HEAD TO REST
# NOW DON'T YOU CRY NO MOOOOOOOOORE

**Comment by Cruix** [comment]
I have removed Preternis chemical and radiation immunity and the way they do not metabolize most reagents. They still flush reagents after 40 cycles and do not faint or mutate from radiation.

In order to do exactly what Adam wants I will have to rewrite the human damage system for a third time, and I really don't have time to do that.

**Comment by Super3222** [comment]
It's alright. It's already 99.9% close to the original design.
But anyway - can we test merge this for the next month.

**Comment by ShadowDeath6** [comment]
I thought the commit said "removed preternis entirely"

was disappointed

**Comment by Super3222** [comment]
I feel like https://github.com/yogstation13/yogstation/issues/1855 is related to this PR.

**Pull request merged**